instant-xml/tests/all.rs

```diff
@@ -41,52 +35,104 @@ fn struct_with_named_fields() {
+            flag_prefix: false,
+            flag_direct_namespace: true,
             test: Nested {
-                flag: true,
+                flag_parent_prefix: true,
+                flag_internal_prefix: false,
             },
         }
         .to_xml()
         .unwrap(),
-        "<StructWithCustomField xmlns=\"URI\" xmlns:bar=\"BAZ\" xmlns:foo=\"BAR\"><Nested><bar:flag>true</bar:flag></Nested></StructWithCustomField>"
-
+        "<StructWithCustomField xmlns=\"URI\" xmlns:bar=\"BAZ\" xmlns:foo=\"BAR\" int_attribute=\"42\"><bar:flag_direct_namespace_same_as_prefix>true</bar:flag_direct_namespace_same_as_prefix><bar:flag_prefix>false</bar:flag_prefix><flag_direct_namespace xmlns=\"DIFFERENT\">true</flag_direct_namespace><Nested xmlns:internal=\"INTERNAL\"><bar:flag_parent_prefix>true</bar:flag_parent_prefix><internal:flag_internal_prefix>false</internal:flag_internal_prefix></Nested></StructWithCustomField>"
     );
 }
 
-#[derive(Debug, Eq, PartialEq, ToXml, FromXml)]
-#[xml(namespace("URI", bar = "BAZ"))]
-struct NestedDe {
-    #[xml(namespace(bar))]
-    flag: bool,
+#[derive(Debug, Eq, PartialEq, ToXml)]
+#[xml(namespace(dar = "BAZ", internal = "INTERNAL"))]
+struct NestedDifferentNamespace {
+    #[xml(namespace(dar))]
+    flag_parent_prefix: bool,
+    #[xml(namespace(internal))]
+    flag_internal_prefix: bool,
 }
 
 #[derive(Debug, Eq, PartialEq, ToXml)]
 #[xml(namespace("URI", bar = "BAZ", foo = "BAR"))]
-struct StructWithCustomFieldWrongPrefix {
-    test: NestedWrongPrefix,
+struct StructChildNamespaces {
+    different_child_namespace: NestedDifferentNamespace,
+    same_child_namespace: Nested,
 }
 
+// Tests:
+// - Different child namespace
+// - The same child namespace
 #[test]
-#[should_panic]
-fn struct_with_custom_field_wrong_prefix() {
-    StructWithCustomFieldWrongPrefix {
-        test: NestedWrongPrefix { flag: true },
-    }
-    .to_xml()
-    .unwrap();
+fn struct_child_namespaces() {
+    assert_eq!(
+        StructChildNamespaces {
+            different_child_namespace: NestedDifferentNamespace {
+                flag_parent_prefix: true,
+                flag_internal_prefix: false,
+            },
+            same_child_namespace: Nested {
+                flag_parent_prefix: true,
+                flag_internal_prefix: false,
+            },
+        }
+        .to_xml()
+        .unwrap(),
+        "<StructChildNamespaces xmlns=\"URI\" xmlns:bar=\"BAZ\" xmlns:foo=\"BAR\"><NestedDifferentNamespace xmlns=\"\" xmlns:internal=\"INTERNAL\"><bar:flag_parent_prefix>true</bar:flag_parent_prefix><internal:flag_internal_prefix>false</internal:flag_internal_prefix></NestedDifferentNamespace><Nested xmlns:internal=\"INTERNAL\"><bar:flag_parent_prefix>true</bar:flag_parent_prefix><internal:flag_internal_prefix>false</internal:flag_internal_prefix></Nested></StructChildNamespaces>"
+    );
+}
+
+#[derive(Debug, Eq, PartialEq, FromXml)]
+#[xml(namespace("URI", bar = "BAZ"))]
+struct NestedDe {
```

Comment:
This is unused, get rid of it?

Reply:
This is used, there had to be some mismatch with tests at some point. It is used in the tests for deserializer.